### ISSUE TITLE
[BUG: mistral-common requires pydantic (2.6.1) not matching with mistralai V1 requires pydantic (>=2.8.2,<2.9.0)

### ISSUE THREAD

**Issue by Wendong-Fan** [mistralai/mistral-common#37]
### Python -VV

```shell
Python 3.10.9 (main, Mar  1 2023, 12:33:47) [Clang 14.0.6 ]
```

### Pip Freeze

```shell
Python 3.10.9 (main, Mar  1 2023, 12:33:47) [Clang 14.0.6 ]
(camel-ai-py3.10) bash-3.2$ pip freeze
accelerate==0.33.0
accessible-pygments==0.0.5
agentops==0.3.7
aiofiles==23.2.1
aiohappyeyeballs==2.3.5
aiohttp==3.10.2
aiosignal==1.3.1
alabaster==0.7.16
altair==5.3.0
annotated-types==0.7.0
anthropic==0.29.2
antlr4-python3-runtime==4.9.3
anyio==4.4.0
appnope==0.1.4
asttokens==2.4.1
async-timeout==4.0.3
attrs==24.2.0
av==12.3.0
azure-core==1.30.2
azure-storage-blob==12.22.0
babel==2.16.0
backoff==2.2.1
backports.tarfile==1.2.0
beautifulsoup4==4.12.3
boto3==1.34.157
botocore==1.34.157
cachetools==5.4.0
-e git+https://github.com/camel-ai/camel.git@1a91fa62c44f48139c33fae296b91ddb9e4a4b79#egg=camel_ai
certifi==2024.7.4
cffi==1.17.0
cfgv==3.4.0
chardet==5.2.0
charset-normalizer==3.3.2
click==8.1.7
cohere==4.57
colorama==0.4.6
coloredlogs==15.0.1
comm==0.2.2
commonmark==0.9.1
contourpy==1.2.1
coverage==7.6.1
cryptography==43.0.0
cssselect==1.2.0
curl_cffi==0.6.2
cycler==0.12.1
dataclasses-json==0.6.7
datasets==2.14.4
debugpy==1.8.5
decorator==5.1.1
Deprecated==1.2.14
diffusers==0.30.0
dill==0.3.7
discord.py==2.4.0
distlib==0.3.8
distro==1.9.0
docker==7.1.0
docstring-parser==0.15
docutils==0.19
docx2txt==0.8
duckduckgo_search==6.2.6
effdet==0.4.1
emoji==2.12.1
environs==9.5.0
et-xmlfile==1.1.0
eval_type_backport==0.2.0
exceptiongroup==1.2.2
executing==2.0.1
fastapi==0.112.0
fastavro==1.9.5
feedfinder2==0.0.4
feedparser==6.0.11
ffmpy==0.4.0
filelock==3.15.4
filetype==1.2.0
firecrawl-py==0.0.20
flatbuffers==24.3.25
fonttools==4.53.1
frozenlist==1.4.1
fsspec==2024.6.1
geojson==2.5.0
google-ai-generativelanguage==0.6.4
google-api-core==2.19.1
google-api-python-client==2.140.0
google-auth==2.33.0
google-auth-httplib2==0.2.0
google-cloud-core==2.4.1
google-cloud-storage==2.18.2
google-crc32c==1.5.0
google-generativeai==0.6.0
google-resumable-media==2.7.2
googleapis-common-protos==1.63.2
googlemaps==4.10.0
gradio==3.50.2
gradio_client==0.6.1
groq==0.5.0
grpcio==1.63.0
grpcio-status==1.62.3
grpcio-tools==1.62.3
h11==0.14.0
h2==4.1.0
hpack==4.0.0
httpcore==1.0.5
httplib2==0.22.0
httpx==0.27.0
huggingface-hub==0.24.5
humanfriendly==10.0
hyperframe==6.0.1
identify==2.6.0
idna==3.7
ijson==3.3.0
imageio==2.34.2
imagesize==1.4.1
importlib-metadata==6.11.0
importlib_resources==6.4.0
iniconfig==2.0.0
iopath==0.1.10
ipykernel==6.29.5
ipython==8.18.1
isodate==0.6.1
jaraco.context==5.3.0
jedi==0.19.1
jieba3k==0.35.1
Jinja2==3.1.4
jiter==0.5.0
jmespath==1.0.1
joblib==1.4.2
jsonschema==4.21.1
jsonschema-path==0.3.3
jsonschema-specifications==2023.12.1
jupyter_client==8.6.2
jupyter_core==5.7.2
kiwisolver==1.4.5
langdetect==1.0.9
layoutparser==0.3.4
lazy-object-proxy==1.10.0
litellm==1.41.1
lxml==5.2.2
Markdown==3.6
MarkupSafe==2.1.5
marshmallow==3.21.3
matplotlib==3.9.1.post1
matplotlib-inline==0.1.7
milvus-lite==2.4.8
mistral_common==1.3.3
mistralai==0.4.2
mock==5.1.0
more-itertools==10.4.0
mpmath==1.3.0
msg-parser==1.2.0
multidict==6.0.5
multiprocess==0.70.15
mypy==1.11.1
mypy-extensions==1.0.0
narwhals==1.3.0
neo4j==5.23.1
nest-asyncio==1.6.0
networkx==3.2.1
newspaper3k==0.2.8
nltk==3.8.1
nodeenv==1.9.1
numpy==1.26.4
oauthlib==3.2.2
olefile==0.47
omegaconf==2.3.0
onnx==1.16.2
onnxruntime==1.15.1
openai==1.40.2
openapi-schema-validator==0.6.2
openapi-spec-validator==0.7.1
opencv-python==4.10.0.84
openpyxl==3.1.5
orjson==3.10.7
packaging==23.2
pandas==2.2.2
parso==0.8.4
pathable==0.4.3
pathlib==1.0.1
pdf2image==1.17.0
pdfminer.six==20231228
pdfplumber==0.11.3
pexpect==4.9.0
pillow==10.4.0
platformdirs==4.2.2
pluggy==1.5.0
portalocker==2.10.1
prance==23.6.21.0
pre-commit==3.8.0
primp==0.5.5
prompt_toolkit==3.0.47
proto-plus==1.24.0
protobuf==4.25.4
psutil==5.9.8
ptyprocess==0.7.0
pure_eval==0.2.3
pyarrow==17.0.0
pyasn1==0.6.0
pyasn1_modules==0.4.0
pycocotools==2.0.8
pycparser==2.22
pydantic==2.6.1
pydantic_core==2.16.2
pydata-sphinx-theme==0.15.4
pydub==0.25.1
PyGithub==2.3.0
Pygments==2.18.0
PyJWT==2.9.0
pymilvus==2.4.5
PyMuPDF==1.24.9
PyMuPDFb==1.24.9
PyNaCl==1.5.0
pyowm==3.3.0
pypandoc==1.13
pyparsing==3.1.2
pypdfium2==4.30.0
PySocks==1.7.1
pyTelegramBotAPI==4.22.0
pytesseract==0.3.10
pytest==7.4.4
pytest-asyncio==0.23.8
pytest-cov==4.1.0
python-dateutil==2.9.0.post0
python-docx==1.1.2
python-dotenv==1.0.1
python-iso639==2024.4.27
python-magic==0.4.27
python-multipart==0.0.9
python-pptx==0.6.23
pytz==2024.1
PyYAML==6.0.2
pyzmq==26.1.0
qdrant-client==1.10.1
rank-bm25==0.2.2
rapidfuzz==3.9.6
recommonmark==0.7.1
redis==5.0.8
referencing==0.35.1
regex==2024.7.24
requests==2.31.0
requests-file==2.1.0
requests-oauthlib==1.3.1
rfc3339-validator==0.1.4
rpds-py==0.20.0
rsa==4.9
ruamel.yaml==0.18.6
ruamel.yaml.clib==0.2.8
ruff==0.4.10
s3transfer==0.10.2
safetensors==0.4.4
scikit-learn==1.5.1
scipy==1.13.1
semantic-version==2.10.0
sentence-transformers==3.0.1
sentencepiece==0.2.0
sgmllib3k==1.0.0
six==1.16.0
slack_sdk==3.31.0
sniffio==1.3.1
snowballstemmer==2.2.0
soundfile==0.12.1
soupsieve==2.5
Sphinx==6.2.1
sphinx-book-theme==1.1.3
sphinxcontrib-applehelp==2.0.0
sphinxcontrib-devhelp==2.0.0
sphinxcontrib-htmlhelp==2.1.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==2.0.0
sphinxcontrib-serializinghtml==2.0.0
stack-data==0.6.3
starlette==0.37.2
sympy==1.13.1
tabulate==0.9.0
termcolor==2.4.0
threadpoolctl==3.5.0
tiktoken==0.7.0
timm==1.0.8
tinysegmenter==0.3
tldextract==5.1.2
tokenizers==0.19.1
toml==0.10.2
tomli==2.0.1
toolz==0.12.1
torch==2.1.0
torchvision==0.16.0
tornado==6.4.1
tqdm==4.66.5
traitlets==5.14.3
transformers==4.44.0
types-colorama==0.4.15.20240311
types-docutils==0.21.0.20240724
types-mock==5.1.0.20240425
types-Pillow==10.2.0.20240520
types-Pygments==2.18.0.20240506
types-regex==2024.7.24.20240726
types-requests==2.32.0.20240712
types-setuptools==71.1.0.20240806
types-tqdm==4.66.0.20240417
typing-inspect==0.9.0
typing_extensions==4.12.2
tzdata==2024.1
ujson==5.10.0
unstructured==0.10.30
unstructured-inference==0.7.11
unstructured.pytesseract==0.3.12
uritemplate==4.1.1
urllib3==2.2.2
uvicorn==0.30.5
virtualenv==20.26.3
wcwidth==0.2.13
websockets==11.0.3
wikipedia==1.4.0
wolframalpha==5.1.3
wrapt==1.16.0
xlrd==2.0.1
XlsxWriter==3.2.0
xmltodict==0.13.0
xxhash==3.4.1
yarl==1.9.4
zipp==3.19.2
```

### Reproduction Steps

1. add ```mistralai = { version = "^1.0.0", optional = true }```
```mistral-common = { version = "^1.0.0", optional = true }``` to poetry toml
2. run `poetry lock`

### Expected Behavior

    And because mistralai (1.0.1) depends on pydantic (>=2.8.2,<2.9.0), mistralai (>=1.0.0,<2.0.0) requires pydantic (>=2.8.2,<2.9.0).
    And because mistral-common (>=1.0.0,<2.0.0) requires pydantic (2.6.1) (2), mistral-common (>=1.0.0,<2.0.0) is incompatible with mistralai (>=1.0.0,<2.0.0)
    So, because camel-ai depends on both mistralai (^1.0.0) and mistral-common (^1.0.0), version solving failed.

**Comment by patrickvonplaten** [comment]
#40 is merged and 1.3.4 is out - let me know if this solves the issue :-)

**Comment by Wendong-Fan** [comment]
Thanks @patrickvonplaten ! The issue has been solved
